### PR TITLE
runtime: limit new runtime features to booleans via ASSERT checks

### PR DESCRIPTION
**Pull request by alyssawilk**
Adding ASSERTS to get[^Boolean] to ensure we don't add any more non-boolean runtime features.

Risk Level: Medium (if someone else is using this paradigm we may have to roll back or exempt theirs)
Testing: existing tests validate
Docs Changes: n/a
Release Notes: n/a
part of #8992

### CODE REVIEW

File: source/common/runtime/runtime_impl.cc [hunk 1]
```diff
@@ -24,9 +24,24 @@
...
+// Before ASSERTS were added to ensure runtime features were restricted to
+// booleans, several integer features were added. isLegacyFeatures exempts
+// existing integer features from the checks until they can be cleaned up.
+bool isLegacyFeature(absl::string_view feature) {
+  return absl::StartsWith(feature, "envoy.reloadable_features.http2_protocol_options.") ||
+         absl::StartsWith(feature, "envoy.reloadable_features.max_re");
+}
+
+bool isRuntimeFeature(absl::string_view feature) {
+  return absl::StartsWith(feature, "envoy.reloadable_features");
```
Comment (lizan) [inline source/common/runtime/runtime_impl.cc]:
```suggestion
  return absl::StartsWith(feature, "envoy.reloadable_features.");
```

File: source/common/runtime/runtime_impl.cc [hunk 2]
```diff
@@ -24,9 +24,24 @@
...
 namespace Envoy {
 namespace Runtime {
+namespace {
+
+// Before ASSERTS were added to ensure runtime features were restricted to
+// booleans, several integer features were added. isLegacyFeatures exempts
+// existing integer features from the checks until they can be cleaned up.
+bool isLegacyFeature(absl::string_view feature) {
+  return absl::StartsWith(feature, "envoy.reloadable_features.http2_protocol_options.") ||
+         absl::StartsWith(feature, "envoy.reloadable_features.max_re");
```
Comment (lizan) [inline source/common/runtime/runtime_impl.cc]: what does this include? comment?

**Comment by alyssawilk** [comment]
CI looks good!

**Comment by alyssawilk** [comment]
@lizan ping?

**Pull request merged**